examples/all-clusters-app/esp32/main/StatusScreen.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file StatusScreen.cpp
+ *
+ * Screen which displays device state and status:
+ *      - status : Operational | Commissionable
+ *      - IP : IPv6 address, UDP port
+ */
+
+#include "StatusScreen.h"
+
+#if CONFIG_HAVE_DISPLAY
+
+#include "esp_log.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+
+#include <lib/support/CodeUtils.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#include <app/server/Server.h>
+#include <lib/core/DataModelTypes.h>
+
+#define MAX_IP4_ADDRESS_STRING 16
+#define MAX_IP6_ADDRESS_STRING 40
```

Comment:
This should be 46 - 8*4 number + 7 separating :'s + null. Defined already in chip::Inet::IPAddress::kMaxStringLength